src/algo/addon/code/src/pupil.h

```diff
@@ -62,7 +62,7 @@ class Pupil : public Node {
 
         for (auto const& fach : json["subjects"]) {
             pupil.requested_subjects.emplace_back(fach["name"]);
-            if (fach.find("mandatory") != fach.end() && fach["mandatory"]) {
+            if (fach.find("mandatory") != fach.end() && fach["mandatory"].is_boolean() && fach["mandatory"]) {
```

Comment:
As far as I remember, the checks are evaluated from left to right, so I think we should first check if the element exists and then access it.
```suggestion
            if (fach.find("mandatory") != fach.end() &&  fach["mandatory"] && fach["mandatory"].is_boolean()) {
```

Reply:
Yes but .find(...) != .end() checks for existence, fach["mandatory"] is casted to a bool which accesses its value and which fails if it is null. By doing the type check before the cast is not done in the null case.

This can also be seen in the CI run, the testcase goes through fine now (and failed before).

Reply:
Interesting, thank you for the explanation 🙂 
Maybe this can be a comment above the if-condition to explain it for the next one.

Reply:
Added an explicit cast, maybe thats clearer ...